x/dex/keeper/fractional_banker.go

```diff
@@ -72,7 +73,14 @@ func (k *FractionalBanker) GetAllFractionalBalances(ctx sdk.Context) (types.Prec
 
 func (k *FractionalBanker) SetFractionalBalance(ctx sdk.Context, address sdk.AccAddress, balances BalanceMap) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.FractionalBalanceKeyPrefix))
-	for denom, amount := range balances {
+	sortedBalances := make([]string, 0, len(balances))
+	for denom := range balances {
+		sortedBalances = append(sortedBalances, denom)
+	}
+	sort.Strings(sortedBalances)
+
+	for _, denom := range sortedBalances {
+		amount := balances[denom]
```

Comment:
Is it possible to get rid of the map completely? Because it's not the first time when we are thinking (or just missing during the review) that working with maps are safe and it's not the last time when a map shoots us in the foot.

I can imagine where this `BalanceMap` will be reused somewhere and will cause the same problems

Reply:
There will be a map linter instead for now: https://github.com/neutron-org/neutron/pull/990